api/v1alpha1/nicclusterpolicy_webhook.go

```diff
@@ -0,0 +1,304 @@
+	if err := json.Unmarshal([]byte(rdmaSharedDevicePluginConfig), &rdmaSharedDevicePluginConfigJSON); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("Config"),
+			dp.Config, "Invalid json of RdmaSharedDevicePluginConfig"+err.Error()))
+		return allErrs
+	}
+
+	// Perform schema validation
+	rdmaSharedDevicePluginSchemaLoader := gojsonschema.NewReferenceLoader(
+		"file://./schema/rdma_shared_device_plugin_schema.json")
+	rdmaSharedDevicePluginConfigJSONLoader := gojsonschema.NewStringLoader(rdmaSharedDevicePluginConfig)
+	result, err := gojsonschema.Validate(rdmaSharedDevicePluginSchemaLoader, rdmaSharedDevicePluginConfigJSONLoader)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("Config"), dp.Config,
+			"Invalid json configuration of rdmaSharedDevicePluginConfig"+err.Error()))
+	} else if result.Valid() {
+		configListInterface := rdmaSharedDevicePluginConfigJSON["configList"]
+		configList, _ := configListInterface.([]interface{})
+		for _, configInterface := range configList {
+			config := configInterface.(map[string]interface{})
+			resourceName := config["resourceName"].(string)
+			if !isValidRdmaSharedDevicePluginResourceName(resourceName) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("Config"),
+					dp.Config, "Invalid Resource name, it must consist of alphanumeric characters, "+
+						"'-', '_' or '.', and must start and end with an alphanumeric character "+
+						"(e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0"+
+						"-9_.]*)?[A-Za-z0-9]')"))
+			}
+		}
+	} else {
+		for _, ResultErr := range result.Errors() {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("Config"), dp.Config, ResultErr.Description()))
+		}
+	}
+	return allErrs
+}
+
+// validate is a helper function to perform validation for IBKubernetesSpec.
+func (ibk *IBKubernetesSpec) validate(fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if !isValidPKeyGUID(ibk.PKeyGUIDPoolRangeStart) || !isValidPKeyGUID(ibk.PKeyGUIDPoolRangeEnd) {
+		if !isValidPKeyGUID(ibk.PKeyGUIDPoolRangeStart) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("pKeyGUIDPoolRangeStart"),
+				ibk.PKeyGUIDPoolRangeStart, "pKeyGUIDPoolRangeStart must be a valid GUID format:"+
+					"xx:xx:xx:xx:xx:xx:xx:xx with Hexa numbers"))
+		}
+		if !isValidPKeyGUID(ibk.PKeyGUIDPoolRangeEnd) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("pKeyGUIDPoolRangeEnd"),
+				ibk.PKeyGUIDPoolRangeEnd, "pKeyGUIDPoolRangeEnd must be a valid GUID format: "+
+					"xx:xx:xx:xx:xx:xx:xx:xx with Hexa numbers"))
+		}
+		return allErrs
+	} else if !isValidPKeyRange(ibk.PKeyGUIDPoolRangeStart, ibk.PKeyGUIDPoolRangeEnd) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("pKeyGUIDPoolRangeEnd"),
+			ibk.PKeyGUIDPoolRangeEnd, "pKeyGUIDPoolRangeEnd must not be greater than 02:FF:FF:FF:FF:FF:FF:FF"))
```

Comment:
The error message is confusing. in `isValidPkeyRange` you check that RangeEnd > RangeStart, not for absolute values

Reply:
good catch, it's from the old validation logic.
Done